p2pd/main.go

```diff
@@ -421,5 +426,5 @@ func main() {
 		go func() { log.Println(http.ListenAndServe(c.MetricsAddress, nil)) }()
 	}
 
-	select {}
+	d.Serve()
```

Comment:
```suggestion
	if err := d.Serve(); err != nil {
		log.Fatal(err)
	}
```

Let's avoid ignoring the error in case the server will return some errors in future.